src/tyr/route_serializer_osrm.cc

```diff
@@ -1616,10 +1588,272 @@ json::ArrayPtr serialize_legs(const google::protobuf::RepeatedPtrField<valhalla:
     // Keep the leg
     output_legs->emplace_back(std::move(output_leg));
     leg++;
+    leg_index++;
   }
   return output_legs;
 }
 
+//=============================================================================
```

Comment:
this is getting large, what do you think about moving it to a private header since its a nice little encapsulated chunk of functionality?

also yeah the cache is good because of osrm style map matching defaulting to break points for everyy intput location, makes for a ton of legs. you might try benchmarking map matching if you are concerned at all about performance